### PR TITLE
MM-42316: Error message when clicking on test connection button without saving

### PR DESCRIPTION
**Pull request by codedsun**
#### Ticket Link

  Fixes https://github.com/mattermost/mattermost-server/issues/19691

#### Related Pull Requests

- Has server changes (please link here)
- Has mobile changes (please link here)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by codedsun** [comment]
@M-ZubairAhmed Please review and initiate the ci cancelled processes.

File: components/admin_console/schema_admin_settings.jsx [hunk 1]
```diff
@@ -383,6 +383,10 @@ export default class SchemaAdminSettings extends React.PureComponent {
 
     buildButtonSetting = (setting) => {
         const handleRequestAction = (success, error) => {
+            if (this.state.saveNeeded !== false) {
```
Comment (M-ZubairAhmed) [inline components/admin_console/schema_admin_settings.jsx:386]: nit : prefer shorter syntax
```suggestion
            if (!this.state.saveNeeded) {
```

**Comment by M-ZubairAhmed** [comment]
@codedsun i am not sure if we just want to show the error message if settings or not saved or use the unsaved changes for testing the connection and later save it. I will redirect to @BenCookie95 @shazm for that.

**Comment by codedsun** [comment]
@BenCookie95 @shazm Please reply when you get time. Thanks

**Review by BenCookie95 — APPROVED** [review]

**Comment by BenCookie95** [comment]
> @codedsun i am not sure if we just want to show just the error message or use the unsaved changes for testing the connection and later save it. I will redirect to @BenCookie95 @shazm for that.

Hmmm, I think an error message is fine for now because this is the behaviour in a number of places in system console right now

**Review by BenCookie95 — APPROVED** [review]

**Comment by codedsun** [comment]
@BenCookie95 @M-ZubairAhmed This is ready to merge.

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by furqanmlk — COMMENTED** [review]
Thanks @codedsun, I still see the error, when I modified the field twice and click test connection without saving.

https://user-images.githubusercontent.com/6909890/159746409-d6970f53-6f1b-4ad8-ba69-4c9fb3527c06.mp4

**Comment by furqanmlk** [comment]
> Hey @furqanmlk. Thanks for the review. i just got a bit confused by the video. Are you trying to say that after you edited the details and clicked on save the error message of `Connection unsuccessfull: Please save your unchanged changes first` should not be visible then?

Hi @codedsun 
Sorry for the delay. 
Here the are steps,
1. Navigate to `File Storage` page
2. Fill out the fields 
3. Click `Test Connection` button 
4. Observe that connection passed 
5. Modify any field and click `Test Connection` button
6. Notice the error

**Comment by codedsun** [comment]
@furqanmlk Yes that's the correct error that you need to save the changes first and then click on test connection. This is the expected behaviour of the PR. Let me know if I am wrong somewhere.

**Review by furqanmlk — APPROVED** [review]
Thanks @codedsun

**Pull request merged**